test/e2e/storage/testsuites/provisioning.go

```diff
@@ -198,6 +199,15 @@ func testProvisioning(input *provisioningTestInput) {
 		input.pvc.Spec.VolumeMode = &block
 		TestDynamicProvisioning(input.testCase, input.cs, input.pvc, input.sc)
 	})
+
+	It("should allow concurrent writes on the same node", func() {
+		if !input.dInfo.Capabilities[CapMultiPODs] {
+			framework.Skipf("Driver %q does not support multiple concurrent pods - skipping", input.dInfo.Name)
+		}
+		input.testCase.SkipWriteReadCheck = true
+		input.testCase.MultiWriteCheck = true
+		TestDynamicProvisioning(input.testCase, input.cs, input.pvc, input.sc)
```

Comment:
I know this function does a lot of common setup, but I think it's going to get harder to understand/read as we add more functionality that only pertains to one test case.

Instead, can we move the setup into a common function, and keep each test-case specific logic in its own function?

Reply:
I'll have a look at that.

Reply:
Why do I always end up cleaning up tests written by others?! :tired_face:

`StorageClassTest.ExpectUnschedulable` was added very recently and now I need to move that logic to the one test case that actually uses it. Just saying.

Reply:
I ended up not touching `StorageClassTest.ExpectUnschedulable`. IMHO it would be better to have that as a separate parameter for those functions which actually use it, but that would have led to even more code churn. It was false in all cases for `TestDynamicProvisioning` and that function now ignores it. Having it at `true` wouldn't make sense, because why run that function in a configuration where the two test pods cannot be scheduled?

What I have done is move the write/read check out of `TestDynamicProvisioning`. Callers of `TestDynamicProvisioning` now decide which additional checks they want to run on the PV, which makes it a bit easier to understand what each testcase really covers. The "run two pods concurrently" check then hooks into `TestDynamicProvisioning` the same way.

The first three commits in this PR are now generic cleanup commits. If you want, I can submit them separately.

Reply:
Thanks for finding a way to do this. I wasn't expecting you to clean up other tests.  I was thinking to wrap setup in a function, change TestDynamicProvisioning to use the function, and then write a new function for your test case.  But this works too.